commands/roles.js

```diff
@@ -1,55 +1,150 @@
-const embed = require("../utils/embed.js");
-const settings = require("../utils/settings.js");
+const { SlashCommandBuilder } = require('@discordjs/builders');
+const embed = require('../utils/embed.js');
+const settings = require('../utils/settings.js');
 
 module.exports = {
-	name: 'roles',
-	description: 'Lists all self-assignable roles.',
-	usage: '[roles]',
-	cooldown: 5,
-	guildOnly: true,
-	execute(message) {
-		const guildSettings = settings.getSettings(message.guild.id);
+	data: new SlashCommandBuilder()
+		.setName('roles')
+		.setDescription('View and manage your roles.')
+		.addSubcommand(subcommand =>
+			subcommand.setName('view')
+				.setDescription('View roles.')
+		)
+		.addSubcommand(subcommand =>
+			subcommand.setName('add')
+				.setDescription('Add a role.')
+				.addRoleOption(option =>
+					option.setName('role')
+						.setDescription('Select a role to add.')
+						.setRequired(true)
+				)
+		)
+		.addSubcommand(subcommand =>
+			subcommand.setName('remove')
+				.setDescription('Remove a role.')
+				.addRoleOption(option =>
+					option.setName('role')
+						.setDescription('Select a role to remove.')
+						.setRequired(true)
+				)
+		),
+	async execute(interaction) {
+		await interaction.deferReply({
+			ephemeral: true
+		});
 
-		if (!guildSettings.role_cmds) {
-			return message.channel.send({ embeds: [embed("❣ **Role commands are disabled.**")] });
-		}
+		const guildSettings = settings.getSettings(interaction.guild.id);
 
-		let roles = Array.from(message.guild.roles.cache.values());
+		let roles = Array.from(interaction.guild.roles.cache.values());
 		let assignableRoles = roles.slice();
 
 		if (guildSettings.role_blacklist) {
-			assignableRoles = assignableRoles.filter(role => !guildSettings.role_blacklist.includes(role.id));
+			assignableRoles = assignableRoles.filter(role => 
+				!guildSettings.role_blacklist.includes(role.id)
+			);
 		}
 
 		assignableRoles = assignableRoles.filter(role => {
-			if (role.name === "@everyone") return false;
+			if (role.name === '@everyone') return false;
 			if (role.managed) return false;
 			return true;
 		});
 
-		const rolesEmbed = embed(`You can assign or remove these roles with the \`${guildSettings.prefix}role\` command.`)
-			.setTitle(`💖 **Here's a list of all the assignable roles:**`);
+		const rolesEmbed = embed('')
+			.setTitle('💖 **Here\'s a list of all the assignable roles:**');
 
 		let roleList = [];
 		assignableRoles.forEach((role) => {
-			roleList.push(`**${role.name}** \u2013 ${role.id}`);
+			roleList.push(role);
```

Comment:
I think `roleList` might just be a clone of `assignableRoles` if we don't modify `roleList` we can just use `assignableRoles` directly